trino/dbapi.py

```diff
@@ -228,10 +266,31 @@ def cursor(self, legacy_primitive_types: bool = None):
         return Cursor(
             self,
             request,
-            # if legacy_primitive_types is not explicitly set in Cursor, take from Connection
+            # if legacy params are not explicitly set in Cursor, take them from Connection
             legacy_primitive_types if legacy_primitive_types is not None else self.legacy_primitive_types
         )
 
+    def _use_legacy_prepared_statements(self):
+        if self.legacy_prepared_statements is not None:
+            return self.legacy_prepared_statements
+
+        value = legacyPreparedStatementsCache.get((self.host, self.port))
+        if value is None:
+            try:
+                query = trino.client.TrinoQuery(
+                    self._create_request(),
+                    # The version() function was introduced in Trino version 352
+                    query="SELECT node_version FROM system.runtime.nodes WHERE coordinator = true")
```

Comment:
the user may not always have access to `system.runtime` in which case the query will fail and our logic here would assume it supports EXECUTE IMMEDIATE.

Reply:
Ugh, I meant to return `False` there.